packages/headless/src/controllers/commerce/recommendations/headless-recommendations.ts

```diff
@@ -0,0 +1,91 @@
+import {CommerceAPIErrorStatusResponse} from '../../../api/commerce/commerce-api-error-response';
+import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {configuration} from '../../../app/common-reducers';
+import {contextReducer as commerceContext} from '../../../features/commerce/context/context-slice';
+import {
+  fetchRecommendation,
+  updateRecommendationSlotId,
+} from '../../../features/commerce/recommendation/recommendation-actions';
+import {recommendationV2Reducer as recommendation} from '../../../features/commerce/recommendation/recommendation-slice';
+import {loadReducerError, slotIdError} from '../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+/**
+ * The `Recommendations` controller exposes a method for retrieving recomendations content in a commerce interface.
+ */
+export interface Recommendation extends Controller {
+  /**
+   * Fetches the recommendations.
+   */
+  refresh(): void;
+
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `ProductListing` controller.
+   */
+  state: RecommendationState;
+}
+
+export interface RecommendationState {
+  headline: string;
+  products: ProductRecommendation[];
+  error: CommerceAPIErrorStatusResponse | null;
+  isLoading: boolean;
+  responseId: string;
+}
+
+export type RecommendationControllerState = Recommendation['state'];
+
+/**
+ * Creates a `ProductListing` controller instance.
+ *
+ * @param engine - The headless commerce engine.
+ * @returns A `ProductListing` controller instance.
+ */
+export function buildRecommendations(
```

Comment:
All other controller follow the same format, and so this one must absolutely follow the same.

First parameter is always the engine, and the second one is always a `prop` , with an `options` property.

And, on `options`, you can pass parameters.

Something like this:

```typescript
interface RecommendationsOptions {
  slotId: string;
}

interface RecommendationsProps {
    options: RecommendationsOptions;
}

buildRecommendations(engine: CommerceEngine, props: RecommendationsProps) {
   // [...]
}
```